src/Protocols/Internal.hs

```diff
@@ -199,8 +203,8 @@ instance C.KnownNat n => Protocol (C.Vec n a) where
 
 -- XXX: Type families with Signals on LHS are currently broken on Clash:
 instance Protocol (CSignal dom a) where
-  type Fwd (CSignal dom a) = CSignal dom a
-  type Bwd (CSignal dom a) = CSignal dom ()
+  type Fwd (CSignal dom a) = Signal dom a
+  type Bwd (CSignal dom a) = ()
```

Comment:
Why this change? why not:
```haskell
type Bwd (CSignal dom a) = Signal dom ()
```
Now `Bwd ()`  and `Bwd (CSignal dom a)` are the same, and there's even less hope to make `Bwd` injective again.

Reply:
I've given up the hope.. If we make an instance for `DSignal` (`CDSignal`? :woozy_face:) it will have the same `Bwd`.

Reply:
IMO its existence would be motivated similar to `CSignal`: sometimes you just want to deal with a non-backpressure thing in `Circuit` context.

In any case, I just remember the original motivation for dropping injectivity: all the protocols having a simple "acknowledge" on their `Bwd` would have to make newtypes, which turned out to be really annoying.

Reply:
In general, you can mix backpressure and latency just fine; it's often a requirement for efficient pipelining without buffers. You'd have the contract that when `ready` deasserts, you can still transfer _n_ more samples for a certain fixed _n_, after which the transfer stalls. For instance the Avalon Streaming interface supports this notion.

I'm not acquainted enough with `clash-protocols` to really say how this relates here; I just noticed a dichotomy being constructed between latencies and backpressure that I think isn't there.